test/addons/callback-scope/test-resolve-async.js

```diff
@@ -4,7 +4,9 @@ const common = require('../../common');
 const assert = require('assert');
 const { testResolveAsync } = require(`./build/${common.buildType}/binding`);
 
+// Checks that resolving promises from C++ works.
+
 let called = false;
 testResolveAsync().then(() => { called = true; });
 
-setTimeout(() => { assert(called); }, common.platformTimeout(50));
+process.on('beforeExit', () => { assert(called); });
```

Comment:
Why not use `common.mustCall`? It uses essentially the same mechanism
https://github.com/nodejs/node/blob/b85baf738eb806edc0e4f7603084df490720330f/test/common/index.js#L428-L429

Reply:
The goal is to have no `MakeCallback()` calls coming from Node.js before the check is run, since those would also run the microtask queue. `process.on('exit')` would be too late for that, `process.on('beforeExit')` should be okay.